controllers/spaceprovisionerconfig/spaceprovisionerconfig_controller_test.go

```diff
@@ -23,204 +25,255 @@ import (
+	})
 
-			// then
-			require.NoError(t, reconcileErr)
-			AssertThat(t, spc, Is(Ready()))
-		})
+	t.Run("is not ready when memory is depleted in one", func(t *testing.T) {
+		// given
+		spc := ModifySpaceProvisionerConfig(blueprintSpc.DeepCopy(), MaxNumberOfSpaces(5), MaxMemoryUtilizationPercent(80))
+
+		r, req, cl := prepareReconcile(t, spc.DeepCopy(),
+			readyToolchainCluster("cluster1"),
+			hosttest.NewToolchainStatus(
+				hosttest.WithMember("cluster1",
+					hosttest.WithSpaceCount(3),
+					hosttest.WithNodeRoleUsage("worker", 90),
+					hosttest.WithNodeRoleUsage("master", 40),
+				),
+			),
+		)
+
+		// when
+		_, reconcileErr := r.Reconcile(context.TODO(), req)
+		require.NoError(t, cl.Get(context.TODO(), runtimeclient.ObjectKeyFromObject(spc), spc))
+
+		// then
+		assert.NoError(t, reconcileErr)
+		AssertThat(t, spc, Is(NotReadyWithReason(toolchainv1alpha1.SpaceProvisionerConfigInsufficientCapacityReason)), Has(ConsumedSpaceCount(3)), Has(ConsumedMemoryUsage(map[string]int{"worker": 90, "master": 40})))
 	})
 
-	// note that this is checking we "jumping 2 steps" from toolchain cluster not being present at all to be present and ready
-	t.Run("becomes ready when the referenced ToolchainCluster appears and is ready", func(t *testing.T) {
+	t.Run("is not ready when memory is depleted in more", func(t *testing.T) {
 		// given
-		spc := NewSpaceProvisionerConfig("spc", test.HostOperatorNs,
-			ReferencingToolchainCluster("cluster1"),
-			WithReadyConditionInvalid(toolchainv1alpha1.SpaceProvisionerConfigToolchainClusterNotFoundReason))
-		r, req, cl := prepareReconcile(t, spc, &toolchainv1alpha1.ToolchainCluster{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      "cluster1",
-				Namespace: test.HostOperatorNs,
-			},
-			Status: toolchainv1alpha1.ToolchainClusterStatus{
-				Conditions: []toolchainv1alpha1.Condition{
-					{
-						Type:   toolchainv1alpha1.ConditionReady,
-						Status: v1.ConditionTrue,
-					},
-				},
-			},
-		})
+		spc := ModifySpaceProvisionerConfig(blueprintSpc.DeepCopy(), MaxNumberOfSpaces(5), MaxMemoryUtilizationPercent(80))
+
+		r, req, cl := prepareReconcile(t, spc.DeepCopy(),
+			readyToolchainCluster("cluster1"),
+			hosttest.NewToolchainStatus(
+				hosttest.WithMember("cluster1",
+					hosttest.WithSpaceCount(3),
+					hosttest.WithNodeRoleUsage("worker", 42),
+					hosttest.WithNodeRoleUsage("master", 90),
+					hosttest.WithNodeRoleUsage("magic", 90),
+				),
+			),
+		)
 
 		// when
 		_, reconcileErr := r.Reconcile(context.TODO(), req)
 		require.NoError(t, cl.Get(context.TODO(), runtimeclient.ObjectKeyFromObject(spc), spc))
 
 		// then
 		assert.NoError(t, reconcileErr)
-		AssertThat(t, spc, Is(Ready()))
+		AssertThat(t, spc,
+			Is(NotReadyWithReason(toolchainv1alpha1.SpaceProvisionerConfigInsufficientCapacityReason)),
+			Has(ConsumedSpaceCount(3)),
+			Has(ConsumedMemoryUsage(map[string]int{"worker": 42, "master": 90, "magic": 90})))
```

Comment:
it would be nice to have this formatting also in other cases where the line became long.